File: springboot-backend/src/main/java/com/wedspot/backend/controller/VendorController.java
package com.wedspot.backend.controller;

import com.wedspot.backend.Model.Entity.Vendor;
import com.wedspot.backend.services.IVendorService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/vendors")
@RequiredArgsConstructor
public class VendorController {

    private final IVendorService vendorService;

    @GetMapping
    public ResponseEntity<List<Vendor>> getAllVendors() {
        return ResponseEntity.ok(vendorService.getAllVendors());
    }

    @GetMapping("/{id}")
    public ResponseEntity<Vendor> getVendorById(@PathVariable Long id) {
        return vendorService.getVendorById(id)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    @GetMapping("/category/{category}")
    public ResponseEntity<List<Vendor>> getVendorsByCategory(@PathVariable String category) {
        return ResponseEntity.ok(vendorService.getVendorsByCategory(category));
    }

    @GetMapping("/premium")
    public ResponseEntity<List<Vendor>> getPremiumVendors() {
        return ResponseEntity.ok(vendorService.getPremiumVendors());
    }

    @PostMapping
    public ResponseEntity<Vendor> createVendor(@RequestBody Vendor vendor) {
        return ResponseEntity.ok(vendorService.createVendor(vendor));
    }
}

File: backend/Controllers/VendorController.cs
using Microsoft.AspNetCore.Mvc;
using backend.Models.Entities;
using backend.Services;

namespace backend.Controllers
{
    [ApiController]
    [Route("api/v1/vendors")]
    public class VendorController : ControllerBase
    {
        private readonly IVendorService _vendorService;

        public VendorController(IVendorService vendorService)
        {
            _vendorService = vendorService;
        }

        [HttpGet]
        public async Task<ActionResult<List<Vendor>>> GetAllVendors()
        {
            var vendors = await _vendorService.GetAllVendors();
            return Ok(vendors);
        }

        [HttpGet("{id}")]
        public async Task<ActionResult<Vendor>> GetVendorById(long id)
        {
            var vendor = await _vendorService.GetVendorById(id);
            if (vendor == null) return NotFound();
            return Ok(vendor);
        }

        [HttpGet("category/{category}")]
        public async Task<ActionResult<List<Vendor>>> GetVendorsByCategory(string category)
        {
            var vendors = await _vendorService.GetVendorsByCategory(category);
            return Ok(vendors);
        }

        [HttpGet("premium")]
        public async Task<ActionResult<List<Vendor>>> GetPremiumVendors()
        {
            var vendors = await _vendorService.GetPremiumVendors();
            return Ok(vendors);
        }

        [HttpPost]
        public async Task<ActionResult<Vendor>> CreateVendor([FromBody] Vendor vendor)
        {
            var createdVendor = await _vendorService.CreateVendor(vendor);
            return Ok(createdVendor);
        }
    }
}
